src/Compiler.ts

```diff
@@ -89,7 +89,10 @@ const defaultCompilerOptions: CompilerOptions = {
 };
 
 export function createStringCompilerProgram(
-    input: string, options: CompilerOptions = defaultCompilerOptions, filePath = "file.ts"): ts.Program {
+    input: string,
+    options: CompilerOptions = defaultCompilerOptions,
+    filePath = "file.ts",
+    extraFiles?: { [filename: string]: string }): ts.Program {
```

Comment:
I kinda dislike this last parameter, I think it would be nicer if you change the type of `input` to be `string | {[fileName: string]: string}`, and then use a typeof check in the function what you are actually doing. (Maybe you could even remove the last parameter - depending on how much that breaks - since you can then do the same with the first parameter).